\documentclass{article}
\usepackage{graphicx}
\usepackage{outlines}
\usepackage{helvet}
\usepackage{microtype}
\usepackage{hyperref}
\usepackage{tabularx}
\usepackage{fullpage}

% \renewcommand{\familydefault}{\sfdefault}
\title{BIOL BC3308: Intro to Microbial Genomics\\
Spring 2024 syllabus}
\date{}

\begin{document}

\maketitle{} \noindent
\textbf{Instructor: } Philippe Chlenski (\texttt{pac@cs.columbia.edu})\\
\textbf{Office hours: } TBA, CSB 506\\
\textbf{Location/time: } Wednesdays 1:10--4:00 PM, QLab Sulzberger Hall

\section*{Course overview}
This course will focus on understanding and implementing basic bioinformatic algorithms and tools to analyze microbial genomes and genomic information. Topics cover a history of genome sequencing methods, local and global alignment methods, sequence annotation tools, de novo genome assembly, multiple sequence alignments, and simple molecular phylogeny. Theoretical lectures will be taught in parallel with labs focused on hands-on analysis of real-world data so that students create tangible and applicable skills. \textbf{Knowledge of a programming language is required to take this course.} Class notes are intended to be self-contained for these topics.

\section*{Prerequisites}
Introductory biology sequence (BIOL BC1500 and BC1502); general chemistry (CHEM BC2001); Genetics (BC2100); Calculus I and II (MATH UN1101; MATH UN1102); MATLAB for Scientists (BIOL BC2500) or equivalent programming experience.

\section*{Learning objectives}
The primary goal of this course is for students to think about asking Biology questions from a computational perspective. By the end of the course, students will be able to: 
\begin{enumerate}
    \item Understand genomic sequencing methods, their tradeoffs, and what they are used for
    \item Describe common bioinformatic file formats and associated analysis methods
    \item Understand and use common bioinformatic tools for microbial genomic applications 
    \item Develop bioinformatic pipelines to analyze real microbial genome data
    \item Formulate independent research questions and methodological approaches based on literature 
\end{enumerate}

\section*{Student assessment measures}
\textbf{Midterm:} 1 midterm halfway through the semester (100 points, 14.2\%)\\
\textbf{Lab reports:} 6 graded lab reports (240 pts, 40 each, 34.3\%)\\
\textbf{Independent project}, broken into discrete steps (300 pts total; 43\%):
\begin{enumerate}
    \item Research question: 20 pts
    \item Literature-based methodology: 40 pts
    \item Bioinformatic pipeline: 240 pts 
    \begin{enumerate}
        \item Report: 60 pts 
        \item Code: 60 pts 
        \item Presentation 1 (related research article): 60 pts
        \item Presentation 2 (results): 60 pts
    \end{enumerate}
\end{enumerate}
\textbf{Class participation:} 60 pts (8.5\%)

\section*{Course materials and affordable access to course texts}
\emph{All students deserve equal access to course texts. For this reason, there is no required textbook for this class.} All relevant readings and lectures will be posted on Courseworks prior to class Bash and Python will be used extensively in this course; as such, all students must have access to a computer that is capable of running Linux/Mac OSX or a provided Linux VM (instructions will be provided for installation as part of the course). Finally, those looking for more Python resources can see \url{https://wiki.python.org/moin/BeginnersGuide/Programmers} for suggested books, websites, and tutorials. Also, office hours are a great time to get specific questions answered. This is time I specifically set aside for you, so use it!

\section*{Assignment policy}
Homework and lab reports should be submitted through your Courseworks account by the following Tuesday at 12:59 pm, before the start of class. 

\section*{Independent project}
In lieu of a final, each student will conduct an independent research project throughout the second half of the semester. Every student will be given access to the same microbial genomics dataset and will form a unique and independent research question. The projectâ€™s first milestone involves searching the literature to find a suitable example of a bioinformatic pipeline and presenting this paper along with your proposed methods. The second milestone includes a full report and the results of your project. While I encourage you to help each other with implementation, the papers and presentations must be your own work. 

\section*{Late policy}
Two lab reports will be given a free extension for up to three days; to exercise this free extension, you must request it by email at least 24 hours before the deadline. Otherwise, no assignments will be accepted if they are over 3 days late. \textbf{No late assignments will be accepted for your independent project report or presentations.}

\section*{Center for Accessibility \& Disability Services (CARDS) Statement}
If you believe you may encounter barriers to the academic environment due to a documented disability or emerging health challenges, please feel free to contact me and/or the Center for Accessibility Resources \& Disability Services (CARDS). Any student with approved academic accommodations is encouraged to contact me during office hours or via email. If you have questions regarding registering a disability or receiving accommodations for the semester, please contact CARDS at (212) 854-4634, cards@barnard.edu, or learn more at barnard.edu/disabilityservices. CARDS is located in 101 Altschul Hall. 

\section*{Honor code}
I encourage you to work with peers during in-class break-out sessions. However, trial and error is the best tutor for becoming a competent programmer and bioinformatician. Feel free to discuss logic amongst each other, but I expect you to use the laboratory assignments as a time to develop your intuition into performing genomic operations. There are infinitely many solutions, and two codes will never look identical. As it pertains to this course, if I suspect that a student has violated the Honor Code, I will forward my concerns to the Dean of Studies. All work for this course must be conducted in accordance with the Barnard Honor Code: \url{https://barnard.edu/honor-code}, established in 1912, updated 2020.

\section*{Wellness}
It is important for everyone to recognize and identify the different pressures, burdens, and stressors you may be facing, whether personal, emotional, physical, financial, mental, or academic. We as a community urge you to make yourself--your own health, sanity, and wellness--your priority throughout this term and your career here. Sleep, exercise, and eating well can all be a part of a healthy regimen to cope with stress. Resources exist to support you in several sectors of your life, and we encourage you to make use of them. Should you have any questions about navigating these resources, please visit these sites:
\begin{itemize}
    \item \url{http://barnard.edu/primarycare}
    \item \url{http://barnard.edu/counseling}
    \item \url{http://barnard.edu/wellwoman/about}
    \item \url{health.columbia.edu/files/healthservices/pdf/alice_Stressbusters_Support_Network.pdf}
\end{itemize}

\section*{Religious holidays}
All conflicts of this class with religious observances must be made known to me in advance. Please let me know of such a conflict as soon as you discover it so that we may make appropriate alternative arrangements.

\section*{Participation policy}
All students must come to class fully prepared and participate actively in class discussions. Participation points are earned by engaging in class activities; examples include actively working on the questions as groups, using technology respectfully, helping others around you, and staying through the entire class duration. If you need to miss class, please notify me at least 1 week in advance, and we can discuss alternatives for participation at that time.

\pagebreak

\section*{Tentative schedule}
\begin{table}[ht]
    \centering
    \begin{tabularx}{\textwidth}{|>{\hsize=.5\hsize}X|>{\hsize=1.5\hsize}X|X|X|}
    \hline
    Week & Lecture topic & Lab topic & Assignment due\\
    \hline
    1 (Jan 17) & Intro to genomes and genomics, history of sequencing technologies, bacterial stuff / central dogma & Lab 1: Working with python (string finding) & Set up computer environment before the first class (instructions to be provided via email)\\
    \hline
    2 (Jan 24) & Intro to databases, online tools, and common file formats & Lab 2: Working with Biopython (database accessing) & Lab 1 report due; Read:  Edwards 2013\\
    \hline
    3 (Jan 31) & Comparing sequences (dot matrices, local and global alignment algorithms) & Lab 3: BLAST / sequence comparisons & Lab 2 report due; Read: HowTo\_BLASTGuide.pdf\\
    \hline
    4 (Feb 7) & Genome assembly + comparison (de Brujin graphs) & Lab 4: Working with unix on command line; read trimming and de novo assembly & Lab 3 report due; Read: Bankevich 2012,  Antipov 2016; Refresh: Linux tutorial\\
    \hline 
    5 (Feb 14) & Distance matrices / tree building (parsimony, neighbor joining, UPGMA, max likelihood) & Lab 5: Phylogenetic trees & Lab 4 report due; Read:  Shakya 2020, Tartoff 2005, Tasang 2017\\
    \hline
    6 (Feb 21) & BWA transformation and short read alignments & Lab 6: BWA and SNP calling & Lab 5 report due; Read: Trapnell 2009, Li 2009\\
    \hline
    7 (Feb 28) & Advanced topics: Annotation and functional enrichment & Exercise: Annotation and other advanced tools & Lab 6 report due; Read: Carattoli\_2014, Zankari\_2012, Bortolaia 2020\\
    \hline
    8 (Mar 6) & MIDTERM & Pipeline assignment overview & \\
    \hline
    9 (Mar 13) & SPRING BREAK & & \\
    \hline
    10 (Mar 20) & Final project overview & Journal club: interpreting pipelines of scientific papers; intro to final project & Pipeline assignment; Read: Brooks 2018, Stoesser 2016, Matamoros 2017\\
    \hline
    11 (Mar 27) & Class does not meet & & Start thinking!\\
    \hline
    12 (Apr 3) & Presentation 1: question and proposed pipeline with literature references & & \\
    \hline
    13 (Apr 10) & Class does not meet & & \\
    \hline
    14 (Apr 17) & Class does not meet & & \\
    \hline
    15 (Apr 24) & Presentation 2: results & & \\ 
    \hline
    \end{tabularx}
    \label{tab:schedule}
\end{table}

\end{document}